src/client/hud.cpp

```diff
@@ -649,10 +649,26 @@ void drawItemStack(video::IVideoDriver *driver,
 		core::rect<s32> oldViewPort = driver->getViewPort();
 		core::matrix4 oldProjMat = driver->getTransform(video::ETS_PROJECTION);
 		core::matrix4 oldViewMat = driver->getTransform(video::ETS_VIEW);
+		core::rect<s32> viewrect = rect;
+		if (clip)
+			viewrect.clipAgainst(*clip);
 		core::matrix4 ProjMatrix;
+		core::matrix4 ViewMatrix;
+		ViewMatrix.buildProjectionMatrixOrthoLH(
+			2.0f * viewrect.getWidth() / rect.getWidth(),
+			2.0f * viewrect.getHeight() / rect.getHeight(),
+			-1, 100);
```

Comment:
```suggestion
			-1.0f, 100.0f);
```
See http://irrlicht.sourceforge.net/docu/classirr_1_1core_1_1_c_matrix4.html#ae4a0618e2da724a26a5d8a201a63d8a5, these should be floats, too.

Reply:
Yes, that's a good thing to make a new pass on this code, it solves old style issues ;)